dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl

```
@@ -154,6 +144,16 @@ RUN chown {{ .user }} /app
 {{- end }}
 {{- end }}
 
+RUN setcap cap_net_raw,cap_setuid+p {{ $beatHome }}/data/elastic-agent-{{ commit_short }}/components/heartbeat && \
+{{- if .linux_capabilities }}
+# Since the beat is stored at the other end of a symlink we must follow the symlink first
+# For security reasons setcap does not support symlinks. This is smart in the general case
+# but in our specific case since we're building a trusted image from trusted binaries this is
+# fine. Thus, we use readlink to follow the link and setcap on the actual binary
+    readlink -f {{ $beatBinary }} | xargs setcap {{ .linux_capabilities }} && \
```

if we are reading a single link, can't we avoid xargs?
```suggestion
   setcap {{ .linux_capabilities }}  $(readlink -f {{ $beatBinary }}) && \
```

---

will test this to be sure but we don't use this for agent